File: student_details_marks.c
#include <stdio.h>

// Global variables to store student data
char name[50];
int roll_number;
float marks;

// Function to enter student details
void enter_details() {
    printf("Enter Name: ");
    scanf("%s", name);
    printf("Enter Roll Number: ");
    scanf("%d", &roll_number);
    printf("Enter Marks: ");
    scanf("%f", &marks);
    printf("Details Saved!\n");
}

// Function to display student details
void display_details() {
    printf("\n--- Student Details ---\n");
    printf("Name: %s\n", name);
    printf("Roll Number: %d\n", roll_number);
    printf("Marks: %.2f\n", marks);
}

int main() {
    int choice;

    // Loop to keep the program running
    while (1) {
        printf("\nPress 1 to Enter Details\n");
        printf("Press 2 to Display Details\n");
        printf("Press 3 to Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        // Conditions to check user input
        if (choice == 1) {
            enter_details();
        } else if (choice == 2) {
            display_details();
        } else if (choice == 3) {
            printf("Exiting program.\n");
            break;
        } else {
            printf("Invalid choice! Please try again.\n");
        }
    }

    return 0;
}

File: student_details_marks.py
# Global variables to store student data
name = ""
roll_number = 0
marks = 0.0

# Function to enter student details
def enter_details():
    global name, roll_number, marks
    name = input("Enter Name: ")
    roll_number = int(input("Enter Roll Number: "))
    marks = float(input("Enter Marks: "))
    print("Details Saved!")

# Function to display student details
def display_details():
    print("\n--- Student Details ---")
    print(f"Name: {name}")
    print(f"Roll Number: {roll_number}")
    print(f"Marks: {marks}")

# Main loop
while True:
    print("\nPress 1 to Enter Details")
    print("Press 2 to Display Details")
    print("Press 3 to Exit")
    
    choice = input("Enter your choice: ")

    # Conditions to check user input
    if choice == '1':
        enter_details()
    elif choice == '2':
        display_details()
    elif choice == '3':
        print("Exiting program.")
        break
    else:
        print("Invalid choice! Please try again.")
